lib/scholar/neighbors/kd_tree.ex

```diff
@@ -593,6 +600,6 @@ defmodule Scholar.Neighbors.KDTree do
         {nearest_neighbors, {node, data, indices, point, distances, visited, i, mode}}
       end
 
-    nearest_neighbors
+    nearest_neighbors |> Nx.devectorize(keep_names: false)
```

Comment:
```suggestion
    Nx.revectorize(nearest_neighbors, input_vectorized_axes, target_shape: ...)
```

here you have to recalculate the target_shape based on input_shape